src/CAN/can_mapping.c

```diff
@@ -32,6 +32,8 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
         }
         uint32_t bitmask = (1UL << length) - 1;
         uint32_t raw_value = (raw_data >> offset) & bitmask;
+
+        /* normalize endian */
         if (mapping->big_endian) {
                 switch (mapping->length) {
```

Comment:
I should have caught this before but I missed it.  If you use a `default` value here it should lead to an `UNREACHABLE` assertion only.  Why?  Because you don't support more than 4 bytes.  Fail fast this way instead of silently masking the error.